velox/connectors/hive/iceberg/tests/IcebergInsertTest.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/iceberg/tests/IcebergTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+namespace facebook::velox::connector::hive::iceberg::test {
+class IcebergInsertTest : public IcebergTestBase {
+ protected:
+  void SetUp() override {
+    IcebergTestBase::SetUp();
+    rowType_ = ROW(
+        {"c1", "c2", "c3", "c4", "c5", "c6", "c7", "c8", "c9", "c10", "c11"},
+        {BIGINT(),
+         INTEGER(),
+         SMALLINT(),
+         BOOLEAN(),
+         REAL(),
+         DECIMAL(18, 5),
+         DECIMAL(38, 5),
+         BOOLEAN(),
+         VARCHAR(),
+         VARBINARY(),
+         ROW({"id", "name"}, {INTEGER(), VARCHAR()})});
+  }
+
+  void test(RowTypePtr rowType, double nullRatio = 0.0) {
+    const auto outputDirectory = exec::test::TempDirectoryPath::create();
+    const auto dataPath = fmt::format("{}", outputDirectory->getPath());
+    constexpr int32_t numBatches = 10;
+    constexpr int32_t vectorSize = 5'000;
+    const auto vectors = createTestData(numBatches, vectorSize, 0.2);
+    auto dataSink =
+        createIcebergDataSink(rowType_, outputDirectory->getPath(), {});
+
+    for (const auto& vector : vectors) {
+      dataSink->appendData(vector);
+    }
+
+    ASSERT_TRUE(dataSink->finish());
+    const auto commitTasks = dataSink->close();
+    createDuckDbTable(vectors);
+    auto splits = createSplitsForDirectory(dataPath);
+    ASSERT_EQ(splits.size(), commitTasks.size());
+    auto plan = exec::test::PlanBuilder().tableScan(rowType_).planNode();
+    assertQuery(plan, splits, fmt::format("SELECT * FROM tmp"));
+  }
+};
+
+TEST_F(IcebergInsertTest, icebergTableWrite) {
```

Comment:
no need to repeat 'iceberg' or 'insert': 

- icebergTableWrite -> basic
- icebergTableWriteMapAndArray -> mapAndArray